### ISSUE TITLE
CA2000 FP with ? or ?? operators on a return;

### ISSUE THREAD

**Issue by cg110** [dotnet/roslyn-analyzers#3873]
#### Analyzer package

[Microsoft.CodeAnalysis.FxCopAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.FxCopAnalyzers)

#### Package Version

v3.3.0-beta2.final

#### Diagnostic ID

CA2000: Dispose objects before losing scope

#### Repro steps

Using VS2019 16.6.4, targetting .net framework 4.7.2

CA2000 seems to firing incorrectly on usages where a ?? or ? test is made, suggesting that it's branch analysis doesn't detect that the object is referenced in the branch, eg:
```
using System;
using System.Collections.Generic;
using System.Diagnostics;

namespace CA2000Test
{
    class Program
    {
        static void Main(string[] args)
        {
        }

        public IDisposable TestCa2000(string path, bool boolA, bool boolB)
        {
            if (boolA)
            {
                IDisposable entry = null;
                if (Dictionary.ContainsKey(path))
                {
                    entry = Dictionary[path];
                }

                return entry ?? new Process();
            }

            return !boolB ? new Process() : Dictionary[path];
        }

        public Dictionary<string, IDisposable> Dictionary { get; set; }
    }
}
```

The returns trigger a CA2000 warning, unless there's something very subtle it's trying to tell me, it appears to be a false positive.

#### Expected behavior
No warning

#### Actual behavior
Program.cs(26,29,26,42): warning CA2000: Call System.IDisposable.Dispose on object created by 'new Process()' before all references to it are out of scope.
Program.cs(23,33,23,46): warning CA2000: Call System.IDisposable.Dispose on object created by 'new Process()' before all references to it are out of scope.

**Comment by cg110** [comment]
Note that 3.0.0 doesn't show this behaviour.

**Comment by mavasani** [comment]
Thanks @cg110 - probably something I regressed. I'll make sure this is addressed before stable 3.3.0 release.
